### PR TITLE
Add documentation for private data source connect using kubernetes

### PR DESCRIPTION
**Pull request by chaudyg**
Add kubernetes deployment documentation for private data source connect, available in Cloud Pro and Advanced in closed preview. 

**What is this feature?**

Add kubernetes deployment documentation for private data source connect, available in Cloud Pro and Advanced in closed preview. These docs should be public, but not indexed or searchable. 

**Who is this feature for?**

Closed preview customers, who will all be given a link to the docs.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

I think this is a reasonable place to put these docs, but this feature is only available in Grafana Cloud, so I wonder if this doc should go into Grafana Cloud documentation...

### CODE REVIEW

**Comment by chri2547** [comment]
@chaudyg - yes, these docs belong in the Cloud repo. Thank you for asking about that.

**Comment by mjseaman** [comment]
Added k8s to the list of options and made it option #1. Otherwise LGTM!

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 1]
```diff
@@ -93,9 +93,33 @@ To set up a private data source connection, follow these steps:
 
 1. Connect to Grafana Cloud using the ssh or the pdc agent in the same directory as your private key, and the certificate and known_hosts file Grafana Labs provided to you.
 
-   There are two options for connecting: SSH, or the PDC Agent Docker image.
+   There are three options for connecting: using Kubernetes, SSH, or the PDC Agent Docker image.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:96]:
```suggestion
   There are three connecting options: Kubernetes, SSH, or the PDC Agent Docker image.
```

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 2]
```diff
@@ -93,9 +93,33 @@ To set up a private data source connection, follow these steps:
...
+   
+   - **Option 1:** Using Kubernetes
 
-   - **Option 1:** Using SSH
+     Create a kubernetes secret with the private key, and the certificate and known_hosts file Grafana Labs provided to you.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:100]:
```suggestion
     Create a Kubernetes secret with the private key and the certificate and known_hosts file Grafana Labs provided.
```

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 3]
```diff
@@ -93,9 +93,33 @@ To set up a private data source connection, follow these steps:
...
+
+     ```
+     $ kubectl create secret generic -n ${NAMESPACE} grafana-pdc-agent \
+     --from-file=key=./${SLUG} \
+     --from-file=known_hosts=./known_hosts \
+     --from-file=cert.pub=./${SLUG}-cert.pub
+     ```
+
+     Generate a kubernetes deployment to deploy the agent.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:109]:
```suggestion
     Generate a Kubernetes deployment to deploy the agent.
```

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 4]
```diff
@@ -93,9 +93,33 @@ To set up a private data source connection, follow these steps:
...
+
+     ```
+     SLUG=${SLUG} PDC_GATEWAY=${PDC_GATEWAY}  NAMESPACE=${NAMESPACE} /bin/sh -c "$(curl -fsSL https://raw.githubusercontent.com/grafana/pdc-agent/main/production/kubernetes/install-agent.sh)"
+     kubectl apply -f deployment.yaml
+     ```
+
+     The environment variables used in the previous commands are as follows:
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:116]:
```suggestion
     The following list contains the environment variables used in the previous commands:
```

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 5]
```diff
@@ -93,9 +93,33 @@ To set up a private data source connection, follow these steps:
...
+
+     - ${PDC_GATEWAY}: The URL of the private data source connect in Grafana Cloud. The Grafana team will give you this URL. The URL follows the format `grafana-private-datasource-connect-<cluster>.grafana.net`
+     - ${SLUG}: The name of the stack you want to connect to your data source. For example, the stack `test.grafana.net` has the slug `test.`
+     - ${NAMESPACE}: The kubernetes namespace for the pdc-agent.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:120]:
```suggestion
     - ${NAMESPACE}: The Kubernetes namespace for the pdc-agent.
```

**Review by chri2547 — DISMISSED** [review]
Added suggestions, thank you!

**Review by dafydd-t — APPROVED** [review]
LGTM!

**Pull request merged**